### PR TITLE
Add qe_test_coverage for Hammer BZ-1695163 and BZ-1760773

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Description:**
1. Added qe_test_coverage for Hammer BZ-1695163 and BZ-1760773.
2. Added an additional check to cover BZ-1421298.
3. Changed some legacy unittest asserts to the pytest asserts.

**Test Results:**
```
(.sat_env) [gtalreja@gtalreja cli]$ pytest test_hammer.py::HammerTestCase
========================================================= test session starts ==========================================================
platform linux -- Python 3.8.5, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo
plugins: mock-1.10.4, services-1.3.1
collecting ... 2020-10-23 13:28:37 - conftest - DEBUG - Collected 3 test cases
collected 3 items

test_hammer.py ...                                                                                                               [100%]
 
================================================ 3 passed, 4 warnings in 197.34 seconds ================================================
```

```
(.sat_env) [gtalreja@gtalreja cli]$ pytest test_repository.py::RepositoryTestCase -k test_positive_upload_content
========================================================= test session starts ==========================================================
platform linux -- Python 3.8.5, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo
plugins: mock-1.10.4, services-1.3.1
collecting ... 2020-10-23 10:28:27 - conftest - DEBUG - Collected 67 test cases
collected 67 items / 65 deselected / 2 selected

test_repository.py ..                                                                                                            [100%]

======================================== 2 passed, 65 deselected, 2 warnings in 162.78 seconds =========================================
```

### CODE REVIEW

File: tests/foreman/cli/test_hammer.py [hunk 1]
```diff
@@ -183,16 +189,81 @@ def test_positive_disable_hammer_defaults(self):
...
             # Verify --organization-id is not required to pass if defaults are set
             result = ssh.command('hammer product list')
-            self.assertEqual(result.return_code, 0)
+            assert result.return_code == 0
             # Verify product list fail without using defaults
             result = ssh.command('hammer --no-use-defaults product list')
-            self.assertNotEqual(result.return_code, 0)
-            self.assertFalse(default_product_name in "".join(result.stdout))
+            assert result.return_code != 0
+            assert not (default_product_name in "".join(result.stdout))
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_hammer.py:196]: `assert default_product_name not in "".join(result.stdout) ` like it  ? non-blocking

File: tests/foreman/cli/test_hammer.py [hunk 2]
```diff
@@ -183,16 +189,81 @@ def test_positive_disable_hammer_defaults(self):
...
+        )
+        # Synchronize custom repository
+        try:
+            Repository.synchronize({'id': custom_repo['id']})
+        except CLIReturnCodeError as err:
+            raise CLIFactoryError(f'Failed to synchronize repository\n {err.msg}')
+
+        result = ssh.command(
+            "hammer erratum list --per-page 5 --fields Issued | awk '!/----/'"
+        ).stdout[:2]
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_hammer.py:257]: Do you need this slice?
It's little puzzling (everyone seeing it will ask "why is that here?"), and doesn't seem to provide any tangible benefit.

File: tests/foreman/cli/test_hammer.py [hunk 3]
```diff
@@ -183,16 +189,81 @@ def test_positive_disable_hammer_defaults(self):
...
+            }
+        )
+        # Synchronize custom repository
+        try:
+            Repository.synchronize({'id': custom_repo['id']})
+        except CLIReturnCodeError as err:
+            raise CLIFactoryError(f'Failed to synchronize repository\n {err.msg}')
+
+        result = ssh.command(
+            "hammer erratum list --per-page 5 --fields Issued | awk '!/----/'"
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_hammer.py:256]: There is DSL implemented for `hammer erratum`, so please use it:

```
robottelo.cli.erratum.Erratum.list()
```

File: tests/foreman/cli/test_hammer.py [hunk 4]
```diff
@@ -183,16 +189,81 @@ def test_positive_disable_hammer_defaults(self):
...
+        result = ssh.command("grep DEBUG /etc/candlepin/candlepin.conf")
+        assert result.return_code == 0
+
+        ssh.command("hammer admin logging --all --level-production")
+        # Verify value of `log4j.logger.org.candlepin` as `WARN`
+        result = ssh.command("grep WARN /etc/candlepin/candlepin.conf")
+        assert result.return_code == 0
+
+    @tier1
+    def test_positive_check_errata_dates(self):
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_hammer.py:229]: I understand bugzilla was assigned to component Hammer, and actually required fix in hammer-related code, but honestly I would expect that in specific component file - `tests/foreman/cli/test_errata.py`.

Similar to bz#1421298, which also is assigned to Hammer, but tests resides in `tests/foreman/cli/test_repository.py`

**Review by mirekdlugosz — COMMENTED** [review]
Please use Robottelo DSL where possible.
I left few more questions/suggestions, but overall, looks good. I''m happy to see hammer component being taken care of!

**Comment by Gauravtalreja1** [comment]
Hi, @mirekdlugosz @omkarkhatavkar Thanks for your quick review/comments, I made the necessary changes, will you mind reviewing this again?

**New Test Results:**
```
(.sat_env) [gtalreja@gtalreja cli]$ pytest test_hammer.py::HammerTestCase -k test_positive_check_debug_log_levels
====================================================================== test session starts =======================================================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo
plugins: ibutsu-1.1.1, services-1.3.1, xdist-1.34.0, mock-3.3.1, forked-1.3.0
collecting ... 2020-11-02 10:24:20 - conftest - DEBUG - Collected 2 test cases
collected 2 items / 1 deselected / 1 selected

test_hammer.py .                                                                                                                                           [100%]

========================================================== 1 passed, 1 deselected, 2 warnings in 25.24s ==========================================================


(.sat_env) [gtalreja@gtalreja cli]$ pytest test_errata.py::ErrataTestCase -k test_positive_check_errata_dates
====================================================================== test session starts =======================================================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo
plugins: ibutsu-1.1.1, services-1.3.1, xdist-1.34.0, mock-3.3.1, forked-1.3.0
collecting ... 2020-11-02 10:00:08 - conftest - DEBUG - Collected 21 test cases
collected 21 items / 20 deselected / 1 selected

test_errata.py .                                                                                                                                           [100%]

==================================================== 1 passed, 20 deselected, 2 warnings in 519.29s (0:08:39) ====================================================
```

File: tests/foreman/cli/test_hammer.py [hunk 5]
```diff
@@ -183,16 +189,81 @@ def test_positive_disable_hammer_defaults(self):
...
+                'content-type': 'yum',
+                'product-id': custom_product['id'],
+                'url': CUSTOM_MODULE_STREAM_REPO_2,
+            }
+        )
+        # Synchronize custom repository
+        try:
+            Repository.synchronize({'id': custom_repo['id']})
+        except CLIReturnCodeError as err:
+            raise CLIFactoryError(f'Failed to synchronize repository\n {err.msg}')
```
[the remark this answers is not in the document]
Comment (mirekdlugosz) [reply tests/foreman/cli/test_hammer.py]: ping. I don't insist on changing that, but I would like to learn rationale behind it.

**Review by mirekdlugosz — APPROVED** [review]
Thanks for updates. My main concerns are resolved. There's one unanswered comment, but it's not blocking.

ACK

**Comment by mirekdlugosz** [comment]
@Gauravtalreja1 looks like now this PR has conflicts on `tests/foreman/cli/test_repository.py`. Please take a look at that and resolve.

**Review by omkarkhatavkar — APPROVED** [review]
ACK. waiting on Travis

File: tests/foreman/cli/test_hammer.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_hammer.py]: Yes TBH, for the repo sync example I referred to some repository sync tests earlier, there I saw this structure of try-except, but I think rarely it will fail/except to sync a custom repo, so I will update this in the test...Thank you!
